docs/DeveloperGuide.md

```diff
@@ -474,9 +476,89 @@ Step 1. placeholder
     * Pros: placeholder
     * Cons: placeholder
 
-### Import/export/checkout features
+### Import feature
+
+#### Implementation
+
+The import mechanism is primarily facilitated by the `DengueHotspotTrackerParser#parseCommand()`,
+`ImportCommandParser#parse()`, `ImportCommand#execute()` and
+`CsvDengueHotspotStorage#readDengueHotspotStorage()` methods.
+
+Given below is an example usage scenario and how the import csv mechanism behaves at each step.
+
+Step 1. The user launches the application.
+
+Step 2. The user executes the `import sampledata.csv` command to import a list of cases from `sampledata.csv`.
+`DengueHotspotTrackerParser#parseCommand()` parses the command and, detecting the `import` command word,
+passes the argument `sampledata.csv` to the `importCommandParser`.
+
+Step 3. `ImportCommandParser#parse()` is called. `sampledata.csv` is checked whether it contains any special characters,
+and ends with `.csv`. `ImportCommand` is constructed, taking in the trimmed filepath `sampledata.csv` as an argument.
+
+Step 4. `ImportCommand#execute()` will get the trimmed filepath and retrieve the list of cases to import from `sampledata.csv`.
+
+Step 5. A `CsvDengueHotspotStorage` is created and `CsvDengueHotspotStorage#readDengueHotspotStorage()` is called,
+importing the valid case list if the format is valid.
 
-DengueHotspotTracker data is stored in a CSV file format.
+The following sequence diagram shows how the import command works:
+
+![ImportSequenceDiagram](images/ImportSequenceDiagram.png)
+
+The following activity diagram summarises what happens when a user executes an Import command:
+
+![ImportActivityDiagram](images/ImportActivityDiagram.png)
+
+#### Design considerations
+
+**Aspect: Filenames and directories accepted**
```

Comment:
For design considerations it might be better to follow the structure of the other features, discussing two alternatives as well as their pros and cons!